source/client/stream_decoder.cc

```diff
@@ -118,20 +118,26 @@ void StreamDecoder::onPoolReady(Envoy::Http::RequestEncoder& encoder,
   stream_info_.upstreamInfo()->upstreamTiming().onFirstUpstreamTxByteSent(
       time_source_); // XXX(oschaaf): is this correct?
   const Envoy::Http::Status status =
-      encoder.encodeHeaders(*request_headers_, request_body_size_ == 0);
+      encoder.encodeHeaders(*request_headers_, false);
   if (!status.ok()) {
     ENVOY_LOG_EVERY_POW_2(error,
                           "Request header encoding failure. Might be missing one or more required "
                           "HTTP headers in {}.",
                           *request_headers_);
   }
-  if (request_body_size_ > 0) {
+  std::string json_body = "";
+  auto json_body_header = request_headers_->get(Envoy::Http::LowerCaseString("x-json-body"));
+  if (json_body_header.empty()) {
+    json_body = json_body_header[0]->value().getStringView();
+  }
+  
+  if (json_body.empty()) {
```

Comment:
this stream_decoder not only handle Jason payload case. Please don't change its behavior by assuming all requests have json_body.